File: backend/routes/fiscal.routes.js
const express = require("express");
const router = express.Router();
const FiscalController = require("../controllers/fiscal.controller");

// Rota para visualizar PDF da nota fiscal via invoice_number
router.get("/nota-fiscal/:invoiceNumber/visualizar", FiscalController.visualizarNotaFiscal);

// Rota para download da nota fiscal via invoice_number
router.get("/nota-fiscal/:invoiceNumber/download", FiscalController.downloadNotaFiscal);

// Rota para listar todas as notas
router.get("/nota-fiscal", FiscalController.listarNotasFiscais);

module.exports = router;

File: micro-services/fiscal-document-service/routes/fiscalRoutes.py
from flask import Blueprint
from controllers.fiscalController import listar_notas_fiscais, visualizar_pdf, download_pdf

fiscal_routes = Blueprint("fiscal_routes", __name__)

fiscal_routes.get("/nota-fiscal")(listar_notas_fiscais)
fiscal_routes.get("/nota-fiscal/<invoice_number>/visualizar")(visualizar_pdf)
fiscal_routes.get("/nota-fiscal/<invoice_number>/download")(download_pdf)

